port/linux/omrvmem.c

```diff
@@ -1174,12 +1180,32 @@ reserve_memory_with_mmap(struct OMRPortLibrary *portLibrary, void *address, uint
 			useBackingSharedFile = FALSE;
 			Trc_PRT_vmem_reserve_doubleMapAPI_not_available(address, byteAmount);
 		}
-	} else if (useBackingFile) {
+	} else if (useBackingSharedTmpFile) {
+		/* Generate a unique temporary filename from template and open the file. */
+		char filename[FILE_NAME_SIZE + 1];
+		snprintf(filename, sizeof(filename), "omrvmem_%09d_XXXXXX", getpid());
+		fd = mkostemp(filename, 0);
+		if (OMRPORT_INVALID_FD != fd) {
+			unlink(filename);
+			/* Set the file size with ftruncate. */
+			if (OMRPORT_INVALID_FD == ftruncate(fd, byteAmount)) {
+				close(fd);
+				fd = OMRPORT_INVALID_FD;
+			}
+		}
+		if (OMRPORT_INVALID_FD == fd) {
+			/* Revert to using an ANONYMOUS mmap. */
+			useBackingSharedTmpFile = FALSE;
+			flags = 0;
+			useBackingFile = set_flags_for_mmap(&flags);
+			Trc_PRT_vmem_reserve_tempfile_not_created(filename, byteAmount);
+		}
+	}
+	if (useBackingFile) {
```

Comment:
I see you corrected the logic error by removing `else` here: a blank line would be welcome separation.

Reply:
Done